### ISSUE TITLE
s&box editor - Components fail to load after application restart.

### ISSUE THREAD

**Issue by CoastalDev** [Facepunch/sbox-issues#6124]
### Describe the bug

[Video Here](https://www.youtube.com/watch?v=PPvB0a7YNuI)

When trying to launch after closing the editor, components fail to load, completely breaking the project.

Game Installation Path:
D:\STEAM\steamapps\common\sbox\

Project Path:
D:\s&box\rogue_game\

Component Path:
D:\s&box\rogue_game\code\Components

I have tried:

1. Reinstalling VS Code.
2. Reinstalling s&box.
3. Verifying Integrity of File Cache (Steam Properties)
4. Re-creating the project and adding the components to any GameObject.
5. Running off of C: drive (where Visual Studio is installed)

I know this may not be a global issue, but having extensive troubles with this. Nobody found a solution in the official s&box discord.

Cheers.

### To Reproduce

Re-creatable for me by:

1. Create any component, with or without programming.
2. Save scene.
3. Close editor
4. Re-Open project.

### Expected behavior

Engine saves component for easy re-use on different GameObjects.

### Additional context

Windows 11
RTX 3060
Ryzen 7 5700x3d
16GB Ram
1x nVME SSD (OS)
1x 1TB SSD (Storage / s&box root location)

**Comment by handsomematt** [comment]
Log.txt?

**Comment by CoastalDev** [comment]
I am currently doing a reinstall, I will provide that ASAP.

**Comment by CoastalDev** [comment]
[Log.log](https://github.com/user-attachments/files/16494850/Log.log)

**Comment by CoastalDev** [comment]
There is a 2nd log created at the exact same time as the previous one.

See below.

[Log.0.log](https://github.com/user-attachments/files/16494861/Log.0.log)

**Comment by handsomematt** [comment]
```
2024/08/05 04:14:44.3191	[PackageLoader/GameMenu] Whitelist Error: One or more errors occurred. (Unknown Assembly System.Console, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a) (System.AggregateException)
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Sandbox.AccessControl.VerifyAssembly(Stream dll, TrustedBinaryStream& outStream, Boolean addToWhitelist) in C:\build\_work\sbox\sbox\engine\Sandbox.Access\AccessControl.cs:line 82	
```

You are trying to use Console.WriteLine so it is erroring

**Comment by CoastalDev** [comment]
That was the problem. Removed it, tried the same steps with closing and re-opening. Seems to work without flaw now. 

Cheers!

**Comment by Tankonyako** [comment]
I encountered the same issue. I made an error in the Razor file, but it didn't show any errors after compiling.
When I relaunched the editor, all components were marked as 'Missing Component.' This seems to be related to custom game resources and prefabs.
I think it would be helpful to display this information in the editor console or as a warning before launching the editor.

If I hadn't found this ticket, I probably wouldn't have been able to find the error.
